src/world.rs

```diff
@@ -646,3 +660,67 @@ impl Default for World {
         }
     }
 }
+
+#[cfg(feature="serialize")]
+/// Structure used to serialize a world using a component group.
+pub struct WorldSerializer<'a, G> {
```

Comment:
Yeah, if I implemented `Serialize` or `Deserialize` on the `World` itself, then it would make a new copy of the world.